custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -393,15 +433,24 @@ function DownloadController($scope, $rootScope, $location, locationHierarchy, lo
                 if (vm.selectedYear < 2020) {
                     vm.selectedYear = new Date().getFullYear();
                 }
+            } else if (vm.isPPRSelected()) {
+                var currentYear  = new Date().getFullYear();
+                vm.selectedYear = vm.selectedYear >= 2019 ? vm.selectedYear : currentYear;
+                vm.selectedFormat = vm.formats[1].id;
+                vm.years = _.filter(vm.yearsCopy, function (year) {
+                    return year.id >= 2019;
+                });
             } else {
                 vm.years = vm.yearsCopy;
             }
             vm.onSelectYear({'id': vm.selectedYear});
             vm.selectedFormat = 'xlsx';
         }
-
+        // set data period to monthly for all
+        vm.selectedDataPeriod = 1
```

Comment:
yeah that's basically 1 - montly 3  - quarterly and if in future they would come with half yearly then we can use 6.

Reply:
reason for not pushing it to constant is we have only single quarterly report if we had more then push to constant for sure

Reply:
probably just value strings instead of number to denote the data period. That would be more clear

Reply:
I had initial though but then then for future perspective if anything half yearly or yearly comes then number would make more sense.

Reply:
are you planning to address this? you are I think converting this to string only in the backend. So good to use the strings only here. It would make it all easier

Reply:
updated